### PR TITLE
feat: call segment.identify on form submissions

### PR DESCRIPTION
**Pull request by dstaley**
🎟️ [Asana Task]()
🔍 [Preview Link](https://react-components-git-{branch-slug}-hashicorp.vercel.app)

---

## Description

This PR adds a call to `segment.identify` on form submissions in `MarketoForm`.

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [ ] Add Asana and Preview links above.
- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/marketo-form/utils.ts [hunk 1]
```diff
@@ -174,3 +174,56 @@ export function calculateDefaultValues(
...
+
+export function segmentIdentify(leadFormFields: Record<string, unknown>) {
+  // This function is wrapped in a try/catch to prevent Segment errors from
+  // interrupting the form submission workflow.
+  try {
+    if (
+      typeof window !== 'undefined' &&
+      window.analytics &&
+      window.analytics.identify &&
+      typeof window.analytics.identify === 'function'
```
Comment (alexcarpenter) [inline packages/marketo-form/utils.ts:186]: There is a new analytics helper we can make use of here if you'd like https://github.com/hashicorp/web-platform-packages/blob/main/packages/analytics/analytics-js-helpers.ts#L7-L16

Comment (dstaley) [reply packages/marketo-form/utils.ts:186]: ooooo

Comment (dstaley) [reply packages/marketo-form/utils.ts]: added [`ca9c056` (#832)](https://github.com/hashicorp/react-components/pull/832/commits/ca9c056f4331698fbd738a59d53a46c7ef42baaf)

**Review by alexcarpenter — APPROVED** [review]
LGTM

**Pull request merged**